algebra/src/fft/polynomial/dense.rs

```diff
@@ -1,11 +1,15 @@
 //! A polynomial represented in coefficient form.
 
 use crate::{get_best_evaluation_domain, DenseOrSparsePolynomial, EvaluationDomain, Evaluations};
-use crate::{serialize::*, Field, FromBytes, PrimeField, ToBytes};
+use crate::{
+    serialize::*, Field, PrimeField, FromBytes, FromBytesChecked, SemanticallyValid, ToBytes,
+};
 use rand::Rng;
 use rayon::prelude::*;
 use std::fmt;
-use std::ops::{Add, AddAssign, Deref, DerefMut, Div, Mul, Neg, Sub, SubAssign};
+use std::io::{Read, Result as IoResult, Write};
+use std::ops::{Add, AddAssign, Deref, DerefMut, Div, Mul, MulAssign, Neg, Sub, SubAssign};
+use num_traits::Zero;
 
 /// Stores a polynomial in coefficient form.
 #[derive(Clone, PartialEq, Eq, Hash, Default, CanonicalSerialize, CanonicalDeserialize)]
```

Comment:
I think that deriving `PartialEq` and `Eq` for `DensePolynomial` leads to the same issue as for `GroupVec` concerning the possible presence of trailing zeros. The problem could be solved in a similar way too.

Reply:
I'll do. See my answer abovee